src/reactviews/pages/TableExplorer/TableExplorerCustomPager.tsx

```diff
@@ -0,0 +1,279 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import type {
+    PaginationMetadata,
+    PaginationService,
+    PubSubService,
+    SlickGrid,
+    Subscription,
+} from "@slickgrid-universal/common";
+import React, { useEffect, useImperativeHandle, useRef, useState } from "react";
+import {
+    ChevronLeftRegular,
+    ChevronRightRegular,
+    ChevronDoubleLeftRegular,
+    ChevronDoubleRightRegular,
+    ArrowSyncRegular,
+} from "@fluentui/react-icons";
+import { Dropdown, Option, Combobox, Button } from "@fluentui/react-components";
+
+import "./TableExplorerCustomPager.css";
+import { locConstants as loc } from "../../common/locConstants";
+
+export interface TableExplorerCustomPagerRef {
+    init: (
+        grid: SlickGrid,
+        paginationService: PaginationService,
+        pubSubService: PubSubService,
+    ) => void;
+    dispose: () => void;
+    renderPagination: () => void;
+}
+
+export interface TableExplorerCustomPagerProps {
+    currentRowCount?: number;
+    onLoadSubset?: (rowCount: number) => void;
+}
+
+const TableExplorerCustomPager = React.forwardRef<
+    TableExplorerCustomPagerRef,
+    TableExplorerCustomPagerProps
+>((props, ref) => {
+    const { currentRowCount, onLoadSubset } = props;
+    const [currentPagination, setCurrentPagination] = useState<PaginationMetadata>(
+        {} as PaginationMetadata,
+    );
+    const [isLeftPaginationDisabled, setIsLeftPaginationDisabled] = useState(false);
+    const [isRightPaginationDisabled, setIsRightPaginationDisabled] = useState(false);
+    const [selectedPageSize, setSelectedPageSize] = useState<string>("100");
```

Comment:
Remove all magic numbers to clarify their purpose.

Reply:
Places like this, magic constants are okay because useState() is effectively setting a variable value, so these are kinda the same as `selectedPageSize = 100`.